esp-mbedtls/src/lib.rs

```diff
@@ -125,6 +141,22 @@ impl<T> Session<T> {
                 return Err(TlsError::OutOfMemory);
             }
 
+            let client_crt =
+                calloc(1, size_of::<mbedtls_x509_crt>() as u32) as *mut mbedtls_x509_crt;
+            if client_crt.is_null() {
+                free(ssl_context as *const _);
```

Comment:
if we end-up here we already successfully allocated memory for `crt` so we need to free that, too